File: Y/javascript/bob/bob.js
(function() {
  'use strict';
  function Bob() {

    this.hey = function(message) {
      if(isSilence(message)) {
        return "Fine, be that way.";
      } else if(isScream(message)) {
        return "Woah, chill out!";
      } else if(isQuestion(message)) {
        return "Sure";
      } else {
        return "Whatever"; 
      }
    };

    var isSilence = function(message){
      return (message == "");
    };
    var isQuestion = function(message){
      return (message.slice(-1) == "?");
    };
    var isScream = function(message){
      return (message.toUpperCase() === message);
    };
  }
  module.exports = Bob;
}());

File: Y/python/bob/bob.py
class Bob():
  def hey(self, message):
    message = Message(message)
    if message.is_query():
      return 'Sure.'
    elif message.is_silent():
      return 'Fine. Be that way.'
    elif message.is_shout():
      return 'Woah, chill out!'
    else:
      return 'Whatever.'

class Message():
  def __init__(self, message):
    self.message = message
  
  def is_silent(self):
    return self.message == ''

  def is_shout(self):
    return self.message == self.message.upper()

  def is_query(self):
    return self.message[-1:] == '?'
